### PR TITLE
Fixed issue with poor compression of script files

### LINKED ISSUE #7899: Lua code not compressed in 1.5.0
**Describe the bug (REQUIRED)**
Lua code not compressed in 1.5.0 Beta

**To Reproduce (REQUIRED)**
1. Build with bob
```
java -jar bob/bob.jar --settings bob/settings/dev_game.project_settings --archive  --texture-compression true --with-symbols --variant debug --platform=js-web --bo bob/releases/dev/web -brhtml bob/releases/dev/web/report.html clean resolve build bundle 
```

**Expected behavior (REQUIRED)**
Lua code compressed

**Defold version (REQUIRED):**
  - 1.5.0 beta 
    editor sha d0b08c6914b4fccc7d97e99ba228ac4eadaaee09
   engine sha d0b08c6914b4fccc7d97e99ba228ac4eadaaee09

**Platforms (REQUIRED):**
 - Platforms: [Web]

**Screenshots (OPTIONAL):**
1.4.8

1.5.0

**Additional context (OPTIONAL):**

### PR DESCRIPTION
**Pull request by JCash**
This fixes an issue where the compression ratio of the script files was less than usual.

Fixes #7899

## PR checklist

* [ ] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ArchiveBuilder.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public boolean getForceCompression() {
 
     public boolean shouldUseCompressedResourceData(byte[] original, byte[] compressed) {
         if (this.getForceCompression())
-            return compressed.length < original.length;
+            return true;
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ArchiveBuilder.java:130]: This code path is only used for unit tests, and allows us to more easily test each scenario.

File: engine/resource/src/providers/provider_zip.cpp [hunk 2]
```diff
@@ -239,7 +239,7 @@ static dmResourceProvider::Result GetFileSize(dmResourceProvider::HArchiveIntern
     return dmResourceProvider::RESULT_NOT_FOUND;
 }
 
-static dmResourceProvider::Result UnpackData(const char* path, dmLiveUpdateDDF::ResourceEntry* entry, const uint8_t* raw_resource, uint32_t raw_resource_size, uint8_t* out_buffer)
+static dmResourceProvider::Result UnpackData(const char* path, dmLiveUpdateDDF::ResourceEntry* entry, uint8_t* raw_resource, uint32_t raw_resource_size, uint8_t* out_buffer)
```
Comment (JCash) [inline engine/resource/src/providers/provider_zip.cpp:242]: By removing the "const", we communicate that the in buffer may be altered.

File: engine/resource/src/providers/provider_zip.cpp [hunk 3]
```diff
@@ -249,6 +249,16 @@ static dmResourceProvider::Result UnpackData(const char* path, dmLiveUpdateDDF::
     uint32_t compressed_size = compressed ? entry->m_CompressedSize : entry->m_Size;
     uint32_t resource_size = entry->m_Size;
 
+    if (encrypted)
```
Comment (JCash) [inline engine/resource/src/providers/provider_zip.cpp:252]: Let's decrypt the data before we decompress it.

**Review by AGulev — APPROVED** [review]
👍

**Pull request merged**